daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -1063,7 +1069,7 @@ private[lf] final class Compiler(
       val env = _env.bindExprVar(param, payloadPos).bindExprVar(choice.argBinder._1, choiceArgPos)
       let(
         env,
-        SBResolveSBUBeginExercise(choice.name, choice.consuming, byKey = false)(
+        SBResolveSBUBeginExercise(choice.name, choice.consuming, byKey = false, ifaceId = ifaceId)(
```

Comment:
is this sufficient for ledger API commands with an interface id rather than just the LF exercise by interface? If not, we should either address this in this PR or in a follow-up.

Reply:
This is just for speedy. This doesn't concern ledger API commands at all?

Reply:
Fixed/inherited choices should always have `byInterface` set, even if initially invoked at template id.
I think this is all working correctly wrt remy's PR to support inherited choice exercise by template in the ledger API.

Reply:
the speedy compiler is also used to compile ledger API commands via `compileCommands`. I _think_ it works correctly but haven’t checked. probably worth tackling at least some minimal tests for this sooner rather than later.

Reply:
My PR is still not merge, so a bit tricky to test. ;)

Reply:
It is actually just merged